src/graphics/material/new_material.rs

```diff
@@ -0,0 +1,247 @@
+        metallic: Option<f32>,
+        color: Option<[f32; 4]>,
+    ) -> Self {
+        Self {
+            main_texture,
+            roughness_texture,
+            normal_texture,
+            roughness,
+            metallic,
+            color,
+        }
+    }
+    
+    /// load_data loads the data specified in self if not present in images and returns them bundled as a materialdata
+    pub fn load_data(
+        self,
+        images: &mut HashMap<String,Arc<Image>>,
+        device: &wgpu::Device,
+        encoder: &mut wgpu::CommandEncoder,
+    ) -> NewMaterialData {
+        NewMaterialData {
+            material_kind: MaterialKind::from(&self),
+            main_texture: self.main_texture.map_or(None, |path| {
+                Some(images.entry(path).or_insert(Image::new_color(device, encoder, path.into()).unwrap()//dont unwrap here
+            ).clone())
+            }),
+            roughness_texture: self.roughness_texture.map_or(None, |path| {
+                Some(images.entry(path).or_insert(Image::new_normal(device, encoder, path.into()).unwrap()//dont unwrap here
+            ).clone())
+            }),
+            normal_texture: self.normal_texture.map_or(None, |path| {
+                Some(images.entry(path).or_insert(Image::new_normal(device, encoder, path.into()).unwrap()//dont unwrap here
+            ).clone())
+            }),
+            roughness: self.roughness,
+            metallic: self.metallic,
+            color: self.color,
+            uniform_buf: None,
+        }
+    }
+}
+
+/// load_material_handles reads all valid NewMaterialHandles from path
+pub fn load_material_handles(path: &str) -> Vec<NewMaterialHandle> {
+    let mut material_handles = Vec::new();
+    for entry in WalkDir::new(path) {
+        if let Some(entry) = entry.ok() {
+            if let Some(bytes) = std::fs::read(entry.path()).ok() {
+                // TODO: read could be smarter
+                if let Some(handle) = ron::de::from_bytes::<NewMaterialHandle>(&bytes).ok() {
+                    material_handles.push(handle);
+                }
+            }
+        }
+    }
+    material_handles
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct PBRMaterialUniform {
+    pub color: Vec4,
+    pub info: Vec4,
+}
+
+unsafe impl Zeroable for PBRMaterialUniform {}
+unsafe impl Pod for PBRMaterialUniform {}
+
+#[test]
+fn test_load_mat_nones() {
+    let dummydata = "NewMaterialHandle(
+            main_texture:None,
+            roughness_texture:None,
+            normal_texture:None,
+            roughness:None,
+            metallic:None,
+            color:None,
+        )";
+    let dummystruct = NewMaterialHandle {
+        main_texture: None,
+        roughness_texture: None,
+        normal_texture: None,
+        roughness: None,
+        metallic: None,
+        color: None,
+    };
+    let buf = ron::de::from_str::<NewMaterialHandle>(dummydata).unwrap();
+    let out = ron::ser::to_string(&buf).unwrap();
```

Comment:
I'll be honest I haven't used `serde`/`ron` yet, and this is a great learning experience for me! 👍

Reply:
It's pretty powerful. We aren't even using the good stuff yet.